idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/branchedTransformationUtils.kt

```diff
@@ -82,6 +88,15 @@ private fun KtExpression?.getWhenConditionSubjectCandidate(): KtExpression? = wh
     else -> null
 }
 
+private fun KtExpression.hasCandidateNameReferenceExpression(): Boolean {
+    val nameReferenceExpression = this as? KtNameReferenceExpression
+        ?: (this as? KtQualifiedExpression)?.selectorExpression as? KtNameReferenceExpression
+        ?: return false
+    val resolved = nameReferenceExpression.mainReference.resolve()
```

Comment:
@t-kameyama This change breaks "if to when" intention because resolve of factory-created elements is not possible. Yet I forbid these checks during "if to when", but you may try to make a nicer fix.

Reply:
See this commit: https://github.com/JetBrains/kotlin/commit/d45e83b27d20cfc4b708ed15d68d6973258e49a3